### ISSUE TITLE
Overview of LMs

### ISSUE THREAD

**Issue by JamesStrabala** [JamesStrabala/Capstone-Resorces#13]
### Title
Language Representation Models: An Overview
### URL
 https://doi.org/10.3390/e23111422
### Summary 
Half a page on RNNs, then another half on a summary of transformers I already know. 
Transfer learning is the general process of having a model pre-train on data that isn't strictly related to the task. Bidirectionalism is enabled by masking input during training, so the model knows to look ahead and behind, or by permuting training sentences. 
Some techniques used by deberta (an improvement on bert) include:
* Disentangling word embedding and position so each word is represented twice.
* Adding positional encoding values after word embeddings have gone through a transformer to emphasize context.
The electra model has two encoders, where one is a generator and one is a discriminator.
The albert model is able to shrink parameter size by shrinking the vocabulary matrix into two smaller ones, and by sharing parameters between layers.
### Key Points 
* Many ways to keep building on transformer technology.
** Masking/permuting for bidirectionality.
** Disentangled embedding
** Enhanced masking.
** Discriminators.
** Parameter reduction.
### Citation

@Article{e23111422,
AUTHOR = {Schomacker, Thorben and Tropmann-Frick, Marina},
TITLE = {Language Representation Models: An Overview},
JOURNAL = {Entropy},
VOLUME = {23},
YEAR = {2021},
NUMBER = {11},
ARTICLE-NUMBER = {1422},
URL = {https://www.mdpi.com/1099-4300/23/11/1422},
PubMedID = {34828119},
ISSN = {1099-4300},
ABSTRACT = {In the last few decades, text mining has been used to extract knowledge from free texts. Applying neural networks and deep learning to natural language processing (NLP) tasks has led to many accomplishments for real-world language problems over the years. The developments of the last five years have resulted in techniques that have allowed for the practical application of transfer learning in NLP. The advances in the field have been substantial, and the milestone of outperforming human baseline performance based on the general language understanding evaluation has been achieved. This paper implements a targeted literature review to outline, describe, explain, and put into context the crucial techniques that helped achieve this milestone. The research presented here is a targeted review of neural language models that present vital steps towards a general language representation model.},
DOI = {10.3390/e23111422}
}
### Repo link
N/A
